### DISCUSSION TITLE
Account object temporarily unpopulated, even though wallet is connected

### DISCUSSION THREAD

**Discussion by calebtuttle** [wevm/wagmi#1795] [Q&A]
Using React, Create React App, wagmi 3.5.

I have a call to `useAccount` in a component that wraps a bunch of other components. I noticed that, when I navigate to a different page, the `account` object is populated for about 400ms, then is null for another 400ms or so, and then is populated again. This results in a flash of a fallback component. My wallet (MetaMask) is connected the whole time.

Does anyone know what might be causing this? Does the wallet automatically disconnect and then reconnect? Is this a problem with the version of wagmi we're using?

Any suggestions are appreciated, thanks!

**Comment by tmm** [comment]
What version of wagmi are you using? Can you put together a minimal reproducible example?

**Reply by calebtuttle** [reply]
Thanks for the quick reply. I've discovered that the problem was because our app was using <a> tags for routing instead of making use of react router, so the page reloaded every time. Sorry for the confusion!

`account.connector` still becomes unpopulated during a refresh though. (I.e., on reload, `account.connector` is populated, then becomes unpopulated, then populated again.) This isn't a big issue for us, but it would be great if the connector could stay populated (if that's possible). I created a simple repo that logs the account object to illustrate: https://github.com/calebtuttle/wagmi-example.

**Reply by tmm** [reply]
Thanks for putting that together! What Connector(s) did you notice this happening with?

**Reply by calebtuttle** [reply]
I've seen it with MetaMaskConnector (with MetaMask) and InjectedConnector (with Brave wallet). Haven't tested the others.
